src/select.jsx

```diff
@@ -369,7 +369,7 @@ var classBase = React.createClass({
           tabIndex={-1}
           aria-hidden={true} >
             {React.Children.map(this.props.children, function (child, index) {
-              return <option key={index} value={child.props.value}>{child.props.children}</option>
+              return <option key={index} value={child.props.value}>{child.props.value}</option>
```

Comment:
@benbayard yes, earlier the `children` were directly passed into the option tag which cause the whole issue. With this PR we will pass down the `value` prop when constructing the `options` list.